### PR TITLE
feat(background media): Create background-media component

### PR DESCRIPTION
**Pull request by ddollive**
### Related Ticket(s)
DDS ticket: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6394
HC Jira ticket: https://jsw.ibm.com/browse/HC-2014

### Description
**This is a work in progress and should not be merged yet**

This adds the initial work on the background-media component (https://github.com/carbon-design-system/carbon-for-ibm-dotcom-website/wiki/Background-media).

### Changelog

**New**

- Adds background-media component.

**Changed**

**Removed**

### CODE REVIEW

File: packages/styles/scss/components/background-media/background-media.scss [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                order: 2;
+            }
+        }
+    }
+
+    .#{$prefix}--background-media-image {
+      object-fit: cover;
+      height: 100%;
+      width: 100%;  
+
```
Comment (proeung) [inline packages/styles/scss/components/background-media/background-media.scss:60]: Looks like there's some extra spacing that needs to be adjusted. Try running `yarn format` or run `git commit` to get some of the format and linting in place.

File: packages/styles/scss/components/background-media/background-media.scss [hunk 2]
```diff
@@ -0,0 +1,88 @@
+//
```
Comment (proeung) [inline packages/styles/scss/components/background-media/background-media.scss:1]: Let's rename the file to `_background-media.scss` to keep it consistent with the other `scss` partial within this `styles` package directory.

File: packages/web-components/src/components/background-media/__stories__/background-media.stories.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    
+  `;
+};
+
+export default {
+  title: 'Components/Background Media',
+  decorators: [
+    story =>
+      html`
+
```
Comment (proeung) [inline packages/web-components/src/components/background-media/__stories__/background-media.stories.ts:55]: Same comment as here - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/6472/files#r658258412

File: packages/web-components/src/components/background-media/background-media.scss [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+// Copyright IBM Corp. 2020, 2021
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+@import '@carbon/ibmdotcom-styles/scss/globals/vars';
+@import 'carbon-components/scss/globals/scss/vendor/@carbon/elements/scss/themes/mixins';
+@import '@carbon/ibmdotcom-styles/scss/components/image/image';
+@import '@carbon/ibmdotcom-styles/scss/components/background-media/background-media';
```
Comment (proeung) [inline packages/web-components/src/components/background-media/background-media.scss:11]: A new line is needed or else this build will return a failed message.

File: packages/web-components/src/components/background-media/background-media.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  gradient_direction = GRADIENT_DIRECTION.RIGHT;
+
+  @property({ attribute: 'mobile-position', reflect: true })
+  mobile_position = MOBILE_POSITION.TOP;
+
+  render() {
+    
+    return html`
+      <div class="${prefix}--background-media">
+        <div class="${prefix}--background-media-container">
```
Comment (proeung) [inline packages/web-components/src/components/background-media/background-media.ts:43]: Feels like some of these classes (eg. `${prefix}--background-media-container`) should be adjusted to be an element class name (eg. `${prefix}--background-media__container`) to follow the BEM structure. @ariellalgilmore Let me know if you agree.

Comment (ariellalgilmore) [reply packages/web-components/src/components/background-media/background-media.ts:43]: just messaged Dennis in slack. Agree to follow BEM structure, but hoping we can remove the `div`'s and use the `DDSImage` render() and handle most the functionality within the styles by using the different property values for gradient and mobile position

**Comment by shixiedesign** [comment]
Please note all the deploy previews failed so **Visual QA is not able to start**.

**Comment by proeung** [comment]
I'll be taking over this issue (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6394), therefore we can close out this PR in favor of the new one that will be opening soon.

**Pull request closed without merging**